frontend/e2e/exercises-page.ts

```diff
@@ -22,4 +22,25 @@ export class ExercisesPage {
     async checkExercisePageIsVisible(exercise: string): Promise<void> {
         await expect(this.page.locator(`mat-toolbar .title:has-text("${exercise}")`).first()).toBeVisible();
     }
+
+    async openFilterBar(): Promise<void> {
+        await this.page.click('mat-expansion-panel-header .mat-expansion-indicator');
+    }
+
+    async updateQuery(query: string): Promise<void> {
+        await this.page.fill('.example-form-field .query-input', query);
+    }
+
+    async checkExerciseIsVisible(query: string): Promise<void> {
+        await expect(this.page.locator(`app-exercises-listing .single-exercise:has-text("${query}")`).first()).toBeVisible();
+    }
+
+    async updateDifficultyRangeCheck(minDifficulty: number, maxDifficulty: number): Promise<void> {
+        await this.page.fill('.minDifficulty-input', minDifficulty.toString());
```

Comment:
**issue:** CSS class names can only have lowercase characters and hyphens - no uppercase letters